edith-step/src/components/ReceiptHandler.js

```diff
@@ -0,0 +1,307 @@
+     * data to the receipt-data servlet to find deals and expiration
+     * dates.
+     *
+     * @param {Event} e change event
+     */
+    this.handleSubmit = async (e) => {
+      e.preventDefault();
+      let price = 0;
+      this.state.items.forEach((item) => {
+        price += item.price * item.quantity;
+      });
+      this.setState({totalPrice: price});
+      const response = await axios({
+        method: 'post',
+        url: '/receipt-data',
+        data: {
+          userId: this.state.userId,
+          storeName: this.state.storeName,
+          date: this.state.date,
+          name: this.state.name,
+          fileUrl: this.state.fileUrl,
+          totalPrice: price,
+          items: this.state.items,
+        },
+      });
+
+      const data = response.data;
+      this.setState((state) => ({
+        deals: data,
+      }));
+    };
+
+    /**
+     * Handles secondary submission to the form. Sends the data
+     * to the store-receipt servlet to be stored in Datastore.
+     * @param {Event} e change event
+     */
+    this.addExpirationAndSubmit = async (e) => {
+      let price = 0;
+      this.state.items.forEach((item) => {
+        price += item.price * item.quantity;
+      });
+      this.setState({totalPrice: price});
+      await axios({
+        method: 'post',
+        url: '/store-receipt',
+        data: {
+          userId: this.state.userId,
+          storeName: this.state.storeName,
+          date: this.state.date,
+          name: this.state.name,
+          fileUrl: this.state.fileUrl,
+          totalPrice: price,
+          items: this.state.items,
+          deals: this.state.deals,
+        },
+      });
+      axios({
+        method: 'post',
+        url: '/user-stats-servlet',
+        data: {
+          itemName: this.state.itemName,
+          itemCategory: this.state.itemCategory,
+          itemPrice: this.state.itemPrice,
+          itemQuantity: this.state.itemQuantity,
+          itemDate: this.getDate(),
+          itemReceiptId: this.state.itemReceiptId,
+          itemCategory: this.state.itemCategory,
+        },
+      }).catch((err) => {
+        console.log(err);
+      });
+      this.setState({});
+    };
+  }
+
+  /** Calls function to get receipt data on mount. */
+  async componentDidMount() {
+    const response = await axios({
+      method: 'get',
+      url: '/receipt-file-handler',
+      responseType: 'json',
+    });
+    const receipt = response.data;
+    const itemList = receipt.items;
+    this.setState((state) => ({
+      userId: receipt.userId,
+      storeName: receipt.storeName,
+      date: receipt.date,
+      name: receipt.name,
+      fileUrl: receipt.fileUrl,
+      totalPrice: receipt.totalPrice,
+      items: itemList,
+      deals: [],
+    }));
+  }
+
+  /**
+   * Handles changes to the name of a grocery item.
+   * @param {number} i index of item
+   * @param {Event} e change event
+   */
+  handleNameChange(i, e) {
```

Comment:
avoid single character function arguments (except in small arrow functions where they are obvious from context or unimportant)